uniffi_bindgen/src/interface/object.rs

```diff
@@ -602,6 +698,29 @@ impl APIConverter<Method> for weedle::interface::OperationInterfaceMember<'_> {
     }
 }
 
+#[derive(Debug, Clone, Checksum)]
+pub struct TraitMethod {
+    pub(super) id: UniffiTrait,
+    // The list of methods exposed by this trait. Most will be 1, but eg, `PartialEq` will have `eq` and `ne`.
+    // The order is defined as being the same order as defined in the rust trait - eg, it's understood that
+    // `eq` is 0, `ne` is 1.
+    pub(super) methods: Vec<Method>,
```

Comment:
I don't love this `methods` field and the hard-coded indexes.  What if `TraitMethod` was an enum?

```
enum TraitMethod {
   Display(Method),
   Debug(Method),
   PartialEq {
       eq: Method,
       ne: Method,
   },
   ...
}
```

I also had some naming suggestions that I don't feel that strongly about:
  - Prefix it with `Uniffi` to make it consistent with `UniffiTrait`
  - s/Method/Methods/

Reply:
> What if TraitMethod was an enum?

I decided to avoid that in the interest if keeping procmacros sane - IIUC, it will need to "emit" these, and emitting an array of MethodMetadata is probably easier than emitting an enum with struct fields?

Happy to do that though if that consideration doesn't actually apply

Reply:
I don't think the enum solution will be more complex for proc-macros. I'm in favor of this change.

Reply:
>s/Method/Methods/

I think the singular "Method" would be more consistent (eg, `Attribute` is singular), but `UniffiTrait` is probably better, yeah